### DISCUSSION TITLE
Restricting Concourse users to execute pipeline to specific environment ( stage / prod / non-prod / QA )

### DISCUSSION THREAD

**Discussion by coolkabir56** [concourse/concourse#7559] [Help & Support]
Can you please help me how I can restrict concourse users having to not to execute the pipeline in the specific environment.
for e.g. :
user 1 : should only execute the concourse pipeline in dev and qa environment
user 2: should only execute the course pipeline in non-pord and dev

**Current Scenario: -**
As of now all the users are able to execute the pipeline in all the environments

**Comment by taylorsilva** [comment]
You might be able to do this by writing your own OPA rules: https://concourse-ci.org/opa.html
